src/map/skill.c

```diff
@@ -7310,14 +7310,18 @@ static int skill_castend_nodamage_id(struct block_list *src, struct block_list *
 				clif->skill_nodamage(src,bl,skill_id,skill_lv,0);
 				break;
 			}
-			if (tsc && tsc->opt1) {
-				status_change_end(bl, SC_FREEZE, INVALID_TIMER);
-				status_change_end(bl, SC_STONE, INVALID_TIMER);
-				status_change_end(bl, SC_SLEEP, INVALID_TIMER);
-				status_change_end(bl, SC_STUN, INVALID_TIMER);
-				status_change_end(bl, SC_WHITEIMPRISON, INVALID_TIMER);
+
+			if (bl->type == BL_PC && tstatus->def_ele != ELE_UNDEAD) {
```

Comment:
you just broke the whole thing.
1. you should only check undead using battle->check_undead otherwise it ignores server configuration
2. by adding the bl->type == BL_PC you broke the skill behavior, it should be able to status recovery non-pc but not on undead property objects